### PR TITLE
Paper: CNN Based ToF Image Processing

### PR DESCRIPTION
**Pull request by tamaslevente**
If you are creating this PR in order to submit a draft of your paper,
see http://procbuild.scipy.org/ for logs generated by the build
process.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#build-your-paper)
for more information.

*Editor*: @cbcunc

*Reviewers*: @cocoaaa, @shubhamsharma1609

### CODE REVIEW

**Comment by scoobies** [comment]
Hi!, I'm @scoobies, a friendly bot.

Type ```@scoobies help``` to discover how I can help you.

**Comment by deniederhut** [comment]
@scoobies assign @cbcunc as editor

**Comment by scoobies** [comment]
Assigned! @cbcunc is now the editor

**Comment by cbcunc** [comment]
Editor’s initial quickie look-over:

- [x] Make sure the paper builds on procbuild
- [x] Make sure that the sections are filled in
- [x] Make sure 8 pages or less exclusive of references

**Comment by deniederhut** [comment]
@scoobies add @cocoaaa to reviewers

**Comment by scoobies** [comment]
@cocoaaa added to the reviewers list!

**Comment by deniederhut** [comment]
@scoobies add @shubhamsharma1609 to reviewers

**Comment by scoobies** [comment]
@shubhamsharma1609 added to the reviewers list!

**Comment by deniederhut** [comment]
We've been having difficulty finding reviewers for some papers, so to make sure everyone has enough time to do their reviews, we will be bumping the deadline for the initial review back to June 17th.

**Review by shubham-scisar — COMMENTED** [review]
Latest commit under review.

**Comment by deniederhut** [comment]
Hey @cocoaaa and @shubhamsharma1609! Have you had a chance to review this paper? We were expecting to see some feedback by yesterday.

**Review by shubham-scisar — CHANGES_REQUESTED** [review]
Overall the concept presented in the paper is really great and the results are presented in a clear manner. In addition,I would like to suggest some changes :

1) In the diagram: **exemplification of ToF camera**, please label shutter on the lens and then the subsequent reflections at the shutter as per the pulse diagram as it will increase readability.

2) Under ** low-level ToF Image processing** subsection, it would be better to include some code as per the description to further exemplify the work.

3) Under the  ** Low-level ToF image pre-processing - Tofsmooth** sub-section, the x-axis of the graphs mention analyzed data is an ambiguous term as there are no units (if you are mentioning analyzed data and by that you mean no.of images) then clearly mention no. of images in brackets as the x-axis title.

4) Please include some example code in the **volumetric estimates sub-section** which gives a clear idea about the implementation.

5) In the conclusion, it is mentioned **In this report**, kindly replace the word **report** with **paper**. Also kindly take care of some grammatical errors(very few) to increase clarity.

It would be great to see some more code snippets in the sub-sections that illustrate the work. Overall the paper seems to fit to be included.

**Comment by shubham-scisar** [comment]
> Hey @cocoaaa and @shubhamsharma1609! Have you had a chance to review this paper? We were expecting to see some feedback by yesterday.

Hi @deniederhut !!!!
Apologies for the delay.I have reviewed the paper and have suggested some changes.

**Comment by tamaslevente** [comment]
Thanks  @shubhamsharma1609 for the constructive review!

@benjaminkelenyi : could you please handle 1?
@molnarszilard : could you please look for 2-3?
@AlexPop104 : please make sure to include some pseudo code (with an appropriate formatting) for 4.
@Funderburger : please have a look for the common terminology through the paper (e.g. paper vs report)

Once you are done, I will proofread the paper again (hopefully this week).

**Comment by deniederhut** [comment]
@scoobies mark needs more review

**Comment by deniederhut** [comment]
@tamaslevente to give everyone more time to respond to their reviews, we've pushed back the deadline for authors' final revisions to July 1st

**Comment by cbcunc** [comment]
@scoobies check references

**Comment by scoobies** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/LRA.2020.2969936 is OK
- 10.1109/ICRA.2011.5980567 is OK
- 10.1109/ICCV.2019.00133 is OK
- 10.1109/CVPR.2017.106 is OK
- 10.1117/12.2586057 is OK
- 10.1177/1729881416657746 is OK
- 10.1109/ACCESS.2020.2982196 is OK

MISSING DOIs

- 10.1109/cvpr.2017.16 may be a valid DOI for title: PointNet: Deep Learning on Point Sets for 3D Classification and Segmentation
- 10.1109/iros45743.2020.9340916 may be a valid DOI for title: Pc-nbv: A point cloud based deep network for efficient next best view planning
- 10.1016/j.rcim.2020.102057 may be a valid DOI for title: Augmented reality integration into MES for connected workers

INVALID DOIs

- https://doi.org/10.1016/j.neucom.2015.06.112 is INVALID because of 'https://doi.org/' prefix
```

**Comment by deniederhut** [comment]
@tamaslevente have you finished addressing the reviewers' comments?

**Comment by deniederhut** [comment]
@scoobies mark pending comment

File: papers/trai/trai.rst [hunk 1]
```diff
@@ -0,0 +1,558 @@
...
+   and deployment. The final deployment targets were Nvidia Xavier NX and AGX platforms receiving data from an Analog Devices ToF camera.
+
+.. class:: keywords
+
+   transfer learning, ToF, python
+
+Introduction
+------------
+   
+Last years the evolution of deep neurThe applications targeted are rangingfrom persond detection, counting, activity analysis to volumetricmeasurements,  mapping  and  navigation  with  mobile  agentsal networks also affected the way in which the Time of Flight (ToF) specific images are processed. The images from the ToF cameras are usually obtained as synchronized depth and infrared (IR) image pairs.
```
Comment (stargaser) [inline papers/trai/trai.rst]: I have a printout of an earlier version that I began marking up. I think some part of this first line has been mixed in with later parts.

```suggestion
In recent years the evolution of deep neural networks has affected the way in which Time of Flight (ToF) images are 
processed. Images from ToF cameras are usually obtained as synchronized depth and infrared (IR) image pairs. 
```

**Comment by tamaslevente** [comment]
> @tamaslevente have you finished addressing the reviewers' comments?

Thanks for the follow up/suggested DOIs. Now each of them seem to be valid.

**Comment by deniederhut** [comment]
Awesome! @shubhamsharma1609 do you feel this paper is now ready for inclusion in the proceedings?

**Comment by shubham-scisar** [comment]
> Awesome! @shubhamsharma1609 do you feel this paper is now ready for inclusion in the proceedings?

@deniederhut I think it is fair enough to go ahead. Approved from my side.

File: papers/trai/.vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (deniederhut) [inline papers/trai/.vscode/settings.json]: Did you mean to commit this?

**Review by tamaslevente — COMMENTED** [review]
@AlexPop104 Please remove this extra vscode dir. Tx!

File: papers/trai/trai.rst [hunk 3]
```diff
@@ -0,0 +1,559 @@
...
+the edge points closest to each other, volume of a box can be computed.
+The advantage of this method is that it allows the computation 
+of the volume for multiple box shaped objects. 
+The following code snippets show the usage of the 
+Sommer's plane determination method to compute the volume.
+
+.. code-block:: python
+    :linenos:
+
+    def compute_volume_ortho(pcd,dmin,dman,votes,seg,thresh):
```
Comment (deniederhut) [inline papers/trai/trai.rst]: So sorry! One more thing -- in the rendered PDF, some of these code lines are extending beyond the column width:

Screen Shot 2021-07-01 at 5 51 04 AM

we might need to make a few changes like, e.g.

```python
def compute_volume_ortho(
    pcd, dmin, dman, votes, seg, thresh
):
```

Comment (tamaslevente) [reply papers/trai/trai.rst]: @AlexPop104 please make some abbreviation in these lines. Tx!

**Comment by cbcunc** [comment]
@scoobies check references

File: papers/trai/trai.rst (hunk 3 above)
Comment (deniederhut) [reply papers/trai/trai.rst]: Thanks!

**Comment by scoobies** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/LRA.2020.2969936 is OK
- 10.1109/ICRA.2011.5980567 is OK
- 10.1109/ICCV.2019.00133 is OK
- 10.1109/CVPR.2017.106 is OK
- 10.1117/12.2586057 is OK
- 10.1109/cvpr.2017.16 is OK
- 10.1016/j.neucom.2015.06.112 is OK
- 10.1177/1729881416657746 is OK
- 10.1109/ACCESS.2020.2982196 is OK
- 10.1109/iros45743.2020.9340916 is OK
- 10.1109/TPAMI.2019.2931577 is OK
- 10.1016/j.rcim.2020.102057 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by cbcunc** [comment]
@scoobies mark ready

**Pull request merged**